From: Torsten Veller <tove@gentoo.org>
Subject: [PATCH] fixes/e3d01d03_Naif_calls_segfault_T_PRTOBJ_of_the_stock_typemap

commit e3d01d03b9fef5b3d7ef1d435ccdd752df78de51
Author: Salvador Ortiz Garcia <sog@msg.com.mx>
Date:   Fri May 28 10:25:52 2010 +0200

    Naif calls segfault T_PRTOBJ of the stock typemap

    The T_PTROBJ INPUT in stock typemap only uses 'sv_derived_from' for
    input validation, that cause a segfault when the argument passed match
    the class name.

---
 lib/ExtUtils/typemap |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/lib/ExtUtils/typemap b/lib/ExtUtils/typemap
index 2c35437..f888587 100644
--- a/lib/ExtUtils/typemap
+++ b/lib/ExtUtils/typemap
@@ -149,7 +149,7 @@ T_REF_IV_PTR
 			${$ALIAS?\q[GvNAME(CvGV(cv))]:\qq[\"$pname\"]},
 			\"$var\", \"$ntype\")
 T_PTROBJ
-	if (sv_derived_from($arg, \"${ntype}\")) {
+	if (SvROK($arg) && sv_derived_from($arg, \"${ntype}\")) {
 	    IV tmp = SvIV((SV*)SvRV($arg));
 	    $var = INT2PTR($type,tmp);
 	}
-- 
tg: (b9ce030..) fixes/e3d01d03_Naif_calls_segfault_T_PRTOBJ_of_the_stock_typemap (depends on: upstream)
